navigation-experimental/src/main/kotlin/com/freeletics/khonshu/navigation/internal/MultiStack.kt

```diff
@@ -147,6 +148,26 @@ internal class MultiStack(
         updateVisibleDestinations()
     }
 
+    fun replaceAll(root: NavRoot) {
+        // remove all stacks
+        val iterator = allStacks.listIterator(allStacks.size)
+        while (iterator.hasPrevious()) {
+            val stack = iterator.previous()
+
+            // Cannot use removeBackStack because we're modifying the list while iterating
+            stack.clear()
+            iterator.remove()
+            onStackEntryRemoved(stack.rootEntry.id)
+        }
```

Comment:
We could do something like this here to avoid duplicating the logic
```suggestion
        while (allStacks.isNotEmpty()) {
            removeBackStack(allStacks.last())
        }
```

Reply:
Done https://github.com/freeletics/khonshu/pull/657/commits/25c159adacb0194d510772c9989d425107ecaac2